marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs

```diff
@@ -120,24 +126,43 @@ utxoStorageTest = property $ do
      >>= liftIO . Storable.getEvents
   Hedgehog.assert (equivalentLists storedEvents events)
 
--- Insert Utxo events in storage, and retrieve the events by address
---
--- utxoInsertAndQueryTest :: Property
--- utxoInsertAndQueryTest = property $ do
---   events <- forAll genUtxoEvents
---   depth <- forAll $ Gen.int (Range.linear 1 5)
---   indexer <- liftIO $ Utxo.open ":memory:" (Utxo.Depth depth)
---              >>= liftIO . Storable.insertMany events
---   let
---     qs :: [StorableQuery Utxo.UtxoHandle]
---     qs = fmap (Utxo.UtxoAddress . Utxo._address) . concatMap (Set.toList . Utxo.ueUtxos) $ events
---   results <- liftIO . traverse (Storable.query Storable.QEverything indexer) $ qs
---   let rows = concatMap (\(Utxo.UtxoResult rs) -> rs ) results
---   computedEvent <-
---     liftIO . Utxo.rowsToEvents (Utxo.getTxIns (getConn indexer) ) $ rows
---   Hedgehog.assert (equivalentLists
---                    computedEvent
---                    (filter (\e -> Utxo.ueChainPoint e /= C.ChainPointAtGenesis) events) )
+---------------- Regression test -------------------
+{-|
+  Insert Utxo events in storage, and retrieve the events
+  The property we're checking here is:
+    - retreived at least one unspent utxo
+    - only retreive unspent utxo's
+    - test `spent` filtering at the boundry of in-memory & disk storage
+-}
+propSaveAndRetrieveUtxoEvents :: Property
+propSaveAndRetrieveUtxoEvents = property $ do
+  -- events <- forAll genUtxoEvents'' -- TODO
```

Comment:
this generator pushes all the spent to the last element of the [events], thus making sure we filter out the in-memory spent. Unfortunately, it meant we weren't testing the SQL LEFT join, therefor I opted out in using it.

Reply:
Why filter out the in-memory spent?

Reply:
I am closing this per our discussion @koslambrou . Also added additional detail to:  [Spec.Marconi.ChainIndex.Indexers.Utxo.UtoxIndex.hs](https://github.com/kayvank/plutus-apps/blob/kayvan/1536-utxo-regression-test/marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs#L90)